### ISSUE TITLE
Possible Implementation Bug in URMA Network Code

### ISSUE THREAD

**Issue by haruki-abe** [nico-bohlinger/one_policy_to_run_them_all#12]
Hi, thank you for sharing this great work.

I found a potential issue in the implementation of the joint attention mechanism in the network.
In the original paper/formulation, the joint attention weight is computed using a softmax over the joint dimension (i.e., across joints), but in the current implementation, it seems the softmax is applied over the channel dimension instead.

This could lead to incorrect weighting and may not match the intended formulation, where attention weights are supposed to be assigned per joint, not per channel.

Example: In `one_policy_to_run_them_all/algorithms/uni_ppo/ppo/policy.py,` line 50 — it seems that the `axis` argument should be `axis=-2` instead of `axis=-1`:

```
joint_e_x = jnp.exp(dynamic_joint_state_mask / (jnp.exp(joint_log_softmax_temperature) + self.softmax_temperature_min))
dynamic_joint_state_mask = joint_e_x / (joint_e_x.sum(axis=-1, keepdims=True) + self.stability_epsilon)
```

If this was not intentional, I would suggest modifying the softmax axis to reflect the correct joint-level attention computation.

Additionally, could you let me know whether this might affect the reported results in the paper?
Would it be necessary to re-run the experiments if the softmax axis was incorrect?

Thanks again for the great work, and I'm looking forward to hearing your thoughts!

**Comment by haruki-abe** [comment]
Thank you very much for your explanation!

From my reading of Equation (2) in the paper, it seems that the softmax and summation are applied over the joint dimension. So the formulation in the paper appears to differ from the current implementation. That’s why I initially thought there might be a mismatch.

Also, I’d like to mention that I have tried modifying the implementation to apply the softmax over the joint dimension as described in Equation (2), and I was able to train the network successfully with similar performance.
This suggests that both approaches may work, though they follow slightly different intuitions.

Thanks again for your great work and I'm looking forward to hearing your thoughts!

**Comment by nico-bohlinger** [comment]
Interesting that it worked for you! I'm currently working on an improved version of the architecture and with my current changes this swap of the sum axis lead to no learning at all anymore. But I agree both variants should in theory work but come from different intuitions / interpretations.

You are totally right, the sum in the denominator of z_j in Equation (2) should be over the latent dimension and not over the joints, to reflect what is actually happening in the code.
Thank you for pointing that out!

**Comment by haruki-abe** [comment]
Thank you for your response!

Your clarification about Equation (2) really helps clear things up.
If possible, it would be very helpful to have an updated version of the paper or an errata in the future. That way, when citing or referring to the method, we can describe more accurately with the intended formulation.

Appreciate your time and openness!
